plugins/modules/docker_plugin.py

```diff
@@ -271,7 +271,7 @@ def update_plugin(self):
                 if not self.check_mode:
                     try:
                         data = prepare_options(self.parameters.plugin_options)
-                        self.client.post_json('/plugins/{0}/set', self.preferred_name, data=['{0}={1}'.format(k, v) for k, v in data.items()])
+                        self.client.post_json('/plugins/{0}/set', self.preferred_name, data=data)
```

Comment:
Given that you are changing the payload contents, it would be Nice to Have (TM) some test to ensure that works.

Reply:
I wrote some tests that I couldn't get to run locally, but apparently they seem to pass in CI... :shrug: